sqlglot/optimizer/annotate_types.py

```diff
@@ -18,11 +22,13 @@ def annotate_types(expression, schema=None, annotators=None, coerces_to=None):
         schema (dict|sqlglot.optimizer.Schema): Database schema.
         annotators (dict): Maps expression type to corresponding annotation function.
         coerces_to (dict): Maps expression type to set of types that it can be coerced into.
+        type_mapping (dict): Maps type (str) to an expression type (DataType). An example:
```

Comment:
i don't think we want type_mapping at all here, it should be part of the schema class and it should leverage sqlglot's parser

Reply:
So, being in schema means that we'd had to `ensure_schema` at the top, correct? Should we add a `get` method to the MappingSchema as well?

Reply:
we should only be dealing with schema objects here yes